### ISSUE TITLE
Update Protobuf submodule

### ISSUE THREAD

**Issue by JamesNK** [grpc/grpc#30259]
The last third_party/protobuf submodule update was to 3.19.4. That was 5 months ago.

Please update it to [21.2](https://github.com/protocolbuffers/protobuf/releases/tag/v21.2) (or later). There are fixes in the proto compiler for C# code gen warnings. They need to be published via `Grpc.Tools` which is built from this repo and uses the submodule.

@yashykt @jtattermusch @apolcyn

**Comment by jtattermusch** [comment]
CC @veblush

**Comment by JamesNK** [comment]
What is happening on this issue?

I think a Protobuf 21.x backport is required for any changes gRPC needs to protobuf so the submodule can be updated for 1.49.x. Then grpc-dotnet can get the fixed `Grpc.Tools` package for that release, and get it out in time for .NET 7.

The protobuf 21.x branch - https://github.com/protocolbuffers/protobuf/tree/21.x - is getting updates. Any required gRPC changes should be backported sooner rather than later to avoid missing the next 21.x patch release.

**Comment by yashykt** [comment]
Hi James, as discussed on the email thread, we'll make sure that 1.49.x takes the next protobuf release. I'll mark this as a release blocker.

**Comment by yashykt** [comment]
The 21.2 release had two issues from gRPC's perspective - 
1) Target ':well_known_protos' is no longer available
2) Target ':protobuf_python' is not publicly visible.
More details at https://github.com/grpc/grpc/pull/30126

Both of these things have been fixed on the 21.x branch (https://github.com/protocolbuffers/protobuf/pull/10118 and https://github.com/protocolbuffers/protobuf/pull/10279), and it seems like 21.3 is going to be tagged today if all goes well.

**Comment by yashykt** [comment]
Updates:
Protobuf released 21.3 and I created a PR to upgrade to it https://github.com/grpc/grpc/pull/30377
On this PR, I noticed that gRPC android and python builds were failing.

The gRPC android failure was due to a protobuf change that required a minimum alignment of 8 instead of 4 for ArenaString. Just sent out a PR to fix that for 21.x - https://github.com/protocolbuffers/protobuf/pull/10298

For python, on chatting with @gnossen, there needed to be some generated file changes that needed to be made that Richard is handling. Additionally, it seems like 21.3 dropped support for python 3.6, so to make this upgrade gRPC python would be dropping support for 3.6 too

**Comment by gnossen** [comment]
I've worked through a few issues integrating with gRPC Python:
- Dropped support for 3.6.
- Updated all protobuf install points to point to the new version
- Fixed a latent bug in the Python test runner where we loaded the same module multiple times under different names
- Regenerated sample _pb2 files.

There's just one more issue I need to work through. It looks like this new version is not working with PyType.
